src/Kiota.Builder/Writers/CSharp/CodeClassDeclarationWriter.cs

```diff
@@ -6,9 +6,13 @@
 namespace Kiota.Builder.Writers.CSharp;
 public class CodeClassDeclarationWriter : BaseElementWriter<ClassDeclaration, CSharpConventionService>
 {
+    private readonly string _clientClassAccessModifier;
     public static string AutoGenerationHeader => "// <auto-generated/>";
     public static string GeneratedCodeAttribute { get; } = $"[global::System.CodeDom.Compiler.GeneratedCode(\"Kiota\", \"{Kiota.Generated.KiotaVersion.Current()}\")]";
-    public CodeClassDeclarationWriter(CSharpConventionService conventionService) : base(conventionService) { }
+    public CodeClassDeclarationWriter(CSharpConventionService conventionService, string? clientClassAccessModifier) : base(conventionService)
+    {
```

Comment:
let's move assigning the default to the generation configuration instead, and make the string non nullable.
And make the initialization from the lock file assign public by default when no value is present (so upgrade is supported)